tests/lib/SimpleSAML/Utils/AttributesTest.php

```diff
@@ -15,33 +16,33 @@ class AttributesTest extends TestCase
     /**
      * Test the getExpectedAttribute() method with invalid attributes array.
      * @return void
+     * @psalm-suppress InvalidArgument
+     * @deprecated Can be removed as soon as the codebase is fully typehinted
      */
     public function testGetExpectedAttributeInvalidAttributesArray()
     {
         // check with empty array as input
         $attributes = 'string';
         $expected = 'string';
-        $this->expectException(
-            \InvalidArgumentException::class,
-            'The attributes array is not an array, it is: '.print_r($attributes, true).'.'
-        );
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('The attributes array is not an array, it is: '.print_r($attributes, true).'.');
         Attributes::getExpectedAttribute($attributes, $expected);
     }
 
 
     /**
      * Test the getExpectedAttributeMethod() method with invalid expected attribute parameter.
+     * @deprecated Remove this test as soon as the codebase is fully typehinted
+     * @psalm-suppress PossiblyFalseArgument
      * @return void
      */
     public function testGetExpectedAttributeInvalidAttributeName()
     {
         // check with invalid attribute name
         $attributes = [];
         $expected = false;
-        $this->expectException(
-            \InvalidArgumentException::class,
-            'The expected attribute is not a string, it is: '.print_r($expected, true).'.'
-        );
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('The expected attribute is not a string, it is: '.print_r($expected, true).'.');
         Attributes::getExpectedAttribute($attributes, $expected);
```

Comment:
Not sure in what PHPunit-version this was allowed, but today it's split in two calls (actually for years already)